php/libraries/UserPermissions.class.inc

```diff
@@ -289,20 +289,29 @@ class UserPermissions
         // create DB object
         $DB =& Database::singleton();
 
-        $query   = "SELECT
+        $query = "SELECT
                 p.permID,
                 p.code,
                 p.description,
                 pc.Description as type,
                 p.action,
                 p.moduleID
-            FROM permissions p
-                JOIN user_perm_rel up ON (p.permID=up.PermID)
+            FROM permissions p ";
+
+        //If not superuser, get the permissions that the user has
+        if ($this->hasPermission('superuser')) {
```

Comment:
I think that line should be:
`if (!$this->hasPermission('superuser')) {`

Then your test should start working again. You can try to add it back. Thanks!